### ISSUE TITLE
Interactive Grid support

### ISSUE THREAD

**Issue by nbuytaert1** [nbuytaert1/apex-select2#73]
Use the **Procedure** API interface instead of the deprecated functions. That's necessary to support Interactive Grid columns.

**Comment by sokosp** [comment]
Hi Nick,
How to use the procedure API? 
we are testing now Sel2 on I.G's.

**Comment by nbuytaert1** [comment]
> How to use the procedure API?
we are testing now Sel2 on I.G's.

I have to change the plugin source code. The current version of the Select2 plugin uses the deprecated **Function** API interface, which doesn't support IG. In the next version of the plugin, I'll change the API interface from functions to procedures.

**Comment by farzadso** [comment]
Just changing it from function to procedure doesn't do the trick Nick, I think some changes need to be made in the code too, because after changing it, it produced this error : PLS-00306: wrong number or types of arguments in call to 'RENDER'

**Comment by rimblas** [comment]
@farzadso, you're correct, check out this discussion: https://community.oracle.com/message/14242489#14242489

**Comment by farzadso** [comment]
@rimblas Thanks Jorge, I guess i'll just wait for the project to be updated whichever method is chosen to overcome the error. Thanks for the interesting link

**Comment by nbuytaert1** [comment]
I got the Select2 plugin working in an IG now, but I'm not really happy with how Select2 items look like in the IG. Here's an example: https://apex.oracle.com/pls/apex/f?p=47839:20

The _department_ column in the _Interactive Grid Support_ region is based on the Select2 plugin. You'll notice that the Select2 items don't integrate very well with the IG.

I guess I'll have to make some CSS and/or JavaScript changes to the Select2 items when being used in an IG. Or has anyone else an interesting idea to better integrate the Select2 items?

**Comment by rimblas** [comment]
Oh interesting, yes it will need a few changes.
No creative ideas here.  Looks like there are a few issues at play, not only CSS related.

**Comment by farzadso** [comment]
@nbuytaert1 Have you placed it in a Text Area. I agree with Jorge @rimblas , there's more than CSS by the looks of it.

**Comment by nbuytaert1** [comment]
I'm getting close to a 100% working solution. I'll release a new version of the plugin by the end of the week.

**Comment by mfreu** [comment]
Great to hear this fantastic News :-)

**Comment by a1b0r** [comment]
There is any news about v3.1.0? What about release?

**Comment by pettinger** [comment]
I could use IG support very well. Any idea when this will be available?

**Comment by nbuytaert1** [comment]
I'm still having a couple of issues with IG support. The Select2 component and IG doesn't play together very well. I don't have that much time work on it, but I hope to release a working solution in one of the next weeks.

**Comment by osydorchuk** [comment]
What about plugin release with IG supports? Maybe you need any help?

**Comment by etaygudai** [comment]
wish too :-)

**Comment by arnoaut** [comment]
have news about the new version?

**Comment by glebovpavel** [comment]
@arnoaut You can try https://github.com/glebovpavel/apex-select2 - this pull request in the main branch is waiting for approval.

**Comment by HeshamAbuElenain** [comment]
Any new news about this bugs? @nbuytaert1

**Comment by pettinger** [comment]
Hi there, i am still waiting for the IG integration. are you still working on the issues? The Position af the select list jumps below the ig as soon as the first values in the list are shown. pls have a look at the attached animated gif
![Select2](https://user-images.githubusercontent.com/20111123/55004249-2abcec80-4fda-11e9-956c-090fcc2d9c39.gif)
